platform/src/naming.rs

```diff
@@ -0,0 +1,119 @@
+
+use regex::Regex;
+use std::fmt;
+use crate::Error;
+
+/// This Enum specifies the types of names that are supported
+pub enum NameKind {
```

Comment:
I'm not super jazzed about this approach to be honest.  It mixes different infrastructure concerns (file system, S3) into a single module. I thinking we'd be better off just making these functions on their respective `StorageProviders` so there is no coupling of concerns across modules.

Reply:
I't gone, functions replaced